### DISCUSSION TITLE
WAVEWATCH III ERROR IN W3BOUNC

### DISCUSSION THREAD

**Discussion by Cesar-IIUNAM** [NOAA-EMC/WW3#1202] [Q&A]
Hi, 
I have generated some spectral points from a "parent" model, and I am trying to impose them as OBCs for the northern boundary of a subdomain (I previously ran this "child" model forced only with winds without problems). However, I am getting the following error:

[ctorresm@master gdm_nest_L1]$ ./ww3_bounc

                  * WAVEWATCH III Bounday input prep. *  
               ===============================================

  Grid name : TEST GRID (GULF OF MEXICO)    

  Comment character is '$'

 Opening file:[ww3.850W230N_201701_spec.nc](http://ww3.850w230n_201701_spec.nc/)
 Opening file:[ww3.855W230N_201701_spec.nc](http://ww3.855w230n_201701_spec.nc/)
 Opening file:[ww3.865W230N_201701_spec.nc](http://ww3.865w230n_201701_spec.nc/)
 Opening file:[ww3.875W230N_201701_spec.nc](http://ww3.875w230n_201701_spec.nc/)
 Opening file:[ww3.885W230N_201701_spec.nc](http://ww3.885w230n_201701_spec.nc/)
 Opening file:[ww3.895W230N_201701_spec.nc](http://ww3.895w230n_201701_spec.nc/)
 Opening file:[ww3.905W230N_201701_spec.nc](http://ww3.905w230n_201701_spec.nc/)
 Opening file:[ww3.915W230N_201701_spec.nc](http://ww3.915w230n_201701_spec.nc/)
 Opening file:[ww3.920W230N_201701_spec.nc](http://ww3.920w230n_201701_spec.nc/)

 * WAVEWATCH III ERROR IN W3BOUNC:
     ILLEGAL NK, NK = 19 DIFFERS FROM NKI = 34
     IT WILL BE MANAGED SOON BY SPCONV

I converted each spectral point into individual ".nc" files, using the program "ww3_ounp." 
Inside my child model, I have "ww3_bounc.inp" configured as follows: 

$ WAVEWATCH III NetCDF boundary input processing                       $
$--------------------------------------------------------------------- $
$
$ Boundary option: READ or WRITE
$
 WRITE
$
$ Interpolation method: 1: nearest
$                       2: linear interpolation
 2
$ Verbose (0, 1, 2)
1
$
$ List of spectra files. These NetCDF files use the WAVEWATCH III
$ format as described in the ww3_ounp.inp file. The files are
$ defined relative to the directory in which the program is run.
$
[ww3.850W230N_201701_spec.nc](http://ww3.850w230n_201701_spec.nc/)
[ww3.855W230N_201701_spec.nc](http://ww3.855w230n_201701_spec.nc/)
[ww3.865W230N_201701_spec.nc](http://ww3.865w230n_201701_spec.nc/)
[ww3.875W230N_201701_spec.nc](http://ww3.875w230n_201701_spec.nc/)
[ww3.885W230N_201701_spec.nc](http://ww3.885w230n_201701_spec.nc/)
[ww3.895W230N_201701_spec.nc](http://ww3.895w230n_201701_spec.nc/)
[ww3.905W230N_201701_spec.nc](http://ww3.905w230n_201701_spec.nc/)
[ww3.915W230N_201701_spec.nc](http://ww3.915w230n_201701_spec.nc/)
[ww3.920W230N_201701_spec.nc](http://ww3.920w230n_201701_spec.nc/)
'STOPSTRING'
$
$ -------------------------------------------------------------------- $
$ End of input file                                                    $
$ -------------------------------------------------------------------- $

In "ww3_grid.inp" of the child model I defined the north border as open, for that, in ww3_grid.inp I typed:

$ Input boundary points from segment data ( FROM = PART ) ------------ $
$

   1  81  F
 141  81  T       (esto lo defino según las dimensiones de mi malla)

I would be very grateful for your help to understand the origin of the error and to be able to give a solution.

Best regards,
César.

**Comment by aliabdolali** [comment]
are spectral resolutions of parent and child setup the same???

**Reply by Cesar-IIUNAM** [reply]
Hello, @aliabdolali ..
thank you for your quick response.

I don't know if I understand your question. The domains have different spatial resolutions, but I think that's not what you mean. 
Although I did the following: the parent model has a spatial resolution of 20 km and in "ww3_shel.inp" I defined spectral points every one degree, spatially distributed over what would be the northern border of my subdomain; which has a resolution of 5 km. 

Please let me know if you need more information. 

César.

**Comment by aliabdolali** [comment]
I meant the number of frequencies and directions, frequency increment and first frequency for both setup. They are defined in your ww3_grid.nml(inp)

**Reply by Cesar-IIUNAM** [reply]
Okay, thank you very much.
They are not the same. 
For the father I have defined:
1.1 0.041 34 72 0.5
For the son:
1.1 0.041 19 24 0.5

Ok, it seems that the problem is in the frequency number between the two models? 19 and 34 and that's why the message "ILLEGAL NK, NK = 19 DIFFERS FROM NKI = 34" ?

**Reply by aliabdolali** [reply]
exactly :) this is the problem.

**Comment by aliabdolali** [comment]
ok, is this difference intentional? I mean in the parern domain, you are considering frequency range of 0.041-0.95 Hz with directional resolution of 5 degrees.
in the child domain, you just consider f=0.041-0.22 Hz with directional resolution of 15 degrees

I can give you pre-processing scripts which does the conversion, if you really want to interpolate.

**Reply by AmbulansAdSomnium** [reply]
Good day @aliabdolali. This situation is typical for me now. I downloaded spectrum data from ERA5. But in my model, different steps and limits in frequency and direction. I would be grateful if you can share the script for preprocessing. Thank you so much.
